atc/worker/image/image_resource_fetcher.go

```diff
@@ -247,8 +262,11 @@ func (i *imageResourceFetcher) ensureVersionOfType(
 		return err
 	}
 
-	checkResourceType := i.resourceFactory.NewResourceForContainer(resourceTypeContainer)
-	versions, err := checkResourceType.Check(context.TODO(), resourceType.Source, nil)
+	processSpec := runtime.ProcessSpec{
+		Path: "/opt/resource/check",
+	}
+	checkResourceType := i.resourceFactory.NewResource(resourceType.Source, nil, resourceType.Version)
+	versions, err := checkResourceType.Check(context.TODO(), processSpec, resourceTypeContainer)
```

Comment:
Valid to make it a real ctx.
However, this has been using ctx.TODO since prior to the refactor - https://github.com/concourse/concourse/blame/master/atc/worker/image/image_resource_fetcher.go#L309